### PR TITLE
[Microsoft.app] Release app microsoft.app 2024 10 02 preview

### PR DESCRIPTION
**Pull request by jijohn14**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by jijohn14** [comment]
We don't have permission to resolve the conflicts with main branch

**Comment by jijohn14** [comment]
Fixes for checks
- https://github.com/Azure/azure-rest-api-specs/pull/31142
- https://github.com/Azure/azure-rest-api-specs/pull/31100

**Comment by TimLovellSmith** [comment]
@jijohn14 The tags on this PR mean its in the state where it should get breaking change approval (first) before ARM approval.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json
Comment (mikekistler) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json:333]: Instead of copying DaprComponent, can you just allOf it?
```suggestion
          "$ref": "./CommonDefinitions.json#/definitions/DaprComponent"
```

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json
Comment (mikekistler) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json:344]: Then you can drop this:
```suggestion
```

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json
Comment (mikekistler) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json:367]: And this can be simplified:
```suggestion
          "provisioningState": {
            "description": "Provisioning state of the Connected Environment Dapr Component.",
            "enum": [
              "Succeeded",
              "Failed",
              "Canceled",
              "InProgress"
            ],
            "type": "string",
            "readOnly": true,
            "x-ms-enum": {
              "name": "ConnectedEnvironmentDaprComponentProvisioningState",
              "modelAsString": true
            }
          },
          "deploymentErrors": {
            "description": "Any errors that occurred during deployment or deployment validation",
            "type": "string",
            "readOnly": true
          }
```

**Review by mikekistler — COMMENTED** [review]
I think you have added the x-ms-flatten in the wrong place. I'm hoping my suggested changes will fix it.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json
Comment (mikekistler) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json:344]: Doesn't work.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json
Comment (mikekistler) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ConnectedEnvironmentsDaprComponents.json:342]: Can we please make one more attempt?
```suggestion
              "$ref": "./CommonDefinitions.json#/definitions/DaprComponentProperties",
              "x-ms-client-flatten": true
```

I am optimistic that this one will work, but if not we'll chalk it up to tooling error and move forward.

**Comment by razvanbadea-msft** [comment]
The first commit needs to be an exact copy of the previous API version. All new changes should only be added in the subsequent commits. 
This allows the reviewer to get a clear understanding of the actual changes being introduced. With the way the PR is raised now, it is not possible for the reviewer to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation. If you are doing the later option please indicate which commit is the exact copy of the previous version.

**Comment by HAKIMBIBI67A** [comment]
Outlook (PWA) Muhammad Ihsan Ali Mail.html aliases Copyright Copy

**Comment by HAKIMBIBI67A** [comment]
> Outlook (PWA) Muhammad Ihsan Ali Mail.html aliases Copyright Copy

ALI MUHAMMAD IHSAN

**Comment by Juliehzl** [comment]
> The first commit needs to be an exact copy of the previous API version. All new changes should only be added in the subsequent commits. This allows the reviewer to get a clear understanding of the actual changes being introduced. With the way the PR is raised now, it is not possible for the reviewer to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation. If you are doing the later option please indicate which commit is the exact copy of the previous version.

@razvanbadea-msft Here is the first commit id [bf061fc8170fc38a9a5a967dd1a0792ae8cf9678 ](https://github.com/Azure/azure-rest-api-specs/commit/bf061fc8170fc38a9a5a967dd1a0792ae8cf9678) which is a copy of previous api version. If you look at code change in our RP Microsoft.App only, you can see all new changes are added in subsequent commits https://github.com/Azure/azure-rest-api-specs/commits/release-app-Microsoft.App-2024-10-02-preview/specification/app/resource-manager starting from bf061fc8170fc38a9a5a967dd1a0792ae8cf9678. In the root folder, you can see many commits which is not related to Microsoft.App because @jijohn14 did a rebase to main branch which introduce the unexpected commits. 

Our PR is open for long time and we are urgent to merge for Ignite. Can you help review with commits https://github.com/Azure/azure-rest-api-specs/commits/release-app-Microsoft.App-2024-10-02-preview/specification/app/resource-manager starting from bf061fc8170fc38a9a5a967dd1a0792ae8cf9678 in Microsoft.App folder only? Thanks a lot. Sorry for inconvenience.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/CommonDefinitions.json
Comment (razvanbadea-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/CommonDefinitions.json:1462]: >deploymentErrors [](http://example.com/codeflow?start=13&length=16)

will this be a string with a concatenated list of errors? or should this be converted in an array?

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/AuthConfigs.json
Comment (razvanbadea-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/AuthConfigs.json:484]: >sasUrlSettingName [](http://example.com/codeflow?start=9&length=17)

is this the actaul sas? if it is a secret you should add the x-ms-secret to it. I did not find it in the example files.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json
Comment (razvanbadea-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json:27]: >v3 [](http://example.com/codeflow?start=69&length=2)

use the v5 also in this file

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json
Comment (razvanbadea-msft) [inline specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json:276]: >int32 [](http://example.com/codeflow?start=21&length=5)

you can also add a minimum and maximum for it

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/CommonDefinitions.json
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/CommonDefinitions.json:1462]: No. It is error string as you can see in example files in https://github.com/Azure/azure-rest-api-specs/pull/29822.

IN addition, this change is added in https://github.com/Azure/azure-rest-api-specs/pull/29822 which already get arm signed off.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/AuthConfigs.json
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/AuthConfigs.json:484]: sasUrlSettingName is a property which already exists in previous api version as you can see in https://github.com/Azure/azure-rest-api-specs/blob/main/specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/AuthConfigs.json#L481. We do not want to update it in new api version.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json:271]: This change is introfuced by https://github.com/Azure/azure-rest-api-specs/pull/31066/commits/5c7cbac44e44bcbfd40e211a6d092d1123220451 with PR https://github.com/Azure/azure-rest-api-specs/pull/30769, which also get arm sign off previously. We do not want to add such limitation in swagger and we prefer validating in RP side for more flexible control.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ManagedEnvironmentsMaintenanceConfigurations.json:276]: same as above

**Comment by Juliehzl** [comment]
@razvanbadea-msft Thanks for reviewing. Actually all changes are reviewed by ARM team previous with separate PRs to the feature branch `release-app-Microsoft.App-2024-10-02-preview`.

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json
Comment (razvanbadea-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json:27]: why in this file you are not using v5?

Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json:27]: For many existing rosource types, we are still using v3. But if v5 is required, I can update to v5. Please help approve  and merge https://github.com/Azure/azure-rest-api-specs/pull/31545. Then it will auto commit to this PR.

**Comment by razvanbadea-msft** [comment]
>       Session Pool is using componentType as the discriminator. While the discriminator is a required property, this rule prevent it being present in the patch request body.

In the session pool the patch is using SessionPoolUpdatableProperties, is this suppression really needed? I do not see a discriminator used in this case

---
Refers to: specification/app/resource-manager/readme.md:86 in 8575f62. [](commit_id = 8575f625529498f48c185132bd2bde74a904ff30, deletion_comment = False)

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json
Comment (razvanbadea-msft) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json:27]: thanks, this helps having an uniform version in all files.

**Comment by HAKIMBIBI67A** [comment]
ALI MUHAMMAD IHSAN

File: specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/ContainerAppsLabelHistory.json]: updated

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**
